testutil/client/tx_broadcast_client.go

```diff
@@ -166,16 +189,23 @@ func (c *BroadcastTxClient) createSendStreams() error {
 }
 
 func (c *BroadcastTxClient) Stop() {
+	c.streamsMapLock.Lock()
+	defer c.streamsMapLock.Unlock()
+
 	totalTxSent := 0
 	for key := range c.streamRoutersMap {
 		if sInfo, ok := c.streamRoutersMap[key]; ok {
+			sInfo.lock.Lock()
 			fmt.Printf("Sent to router %s: txs %d\n", sInfo.endpoint, sInfo.totalTxSent)
 			totalTxSent += int(sInfo.totalTxSent)
 			if sInfo.stream != nil {
 				sInfo.stream.CloseSend()
 			}
+			sInfo.lock.Unlock()
 		}
 	}
 
+	c.streamRoutersMap = nil
+
 	fmt.Printf("Total sent by all routers: txs %d\n", totalTxSent)
```

Comment:
can we pass a logger to this client instead of printing like this? 
maybe in a different PR